File: deprecated/android/repository/model/Speaker.java
package lv.rigadevday.android.repository.model;

import java.util.List;

/**
 */
public class Speaker {

    public int id;
    public String name;
    public String company;
    public String title;
    public String country;
    public String img;
    public String twitter;
    public String blog;
    public String linkedin;
    public String description;

    public Boolean isInList(List<Integer> speakersIds) {
        return speakersIds != null && speakersIds.contains(id);
    }
}

File: app/src/main/kotlin/lv/rigadevday/android/repository/model/Speaker.kt
package lv.rigadevday.android.repository.model

import com.google.firebase.database.IgnoreExtraProperties

@IgnoreExtraProperties
data class Speaker(
    var id: Int = -1,
    var featured: Boolean = false,

    var name: String = "",
    var shortBio: String = "",
    var bio: String = "",
    var photoUrl: String = "",

    var title: String = "",
    var company: String = "",
    var country: String = "",

    var badges: List<Badge> = emptyList(),
    var socials: List<SocialAccount> = emptyList(),
    var tags: List<String> = emptyList()
)

@IgnoreExtraProperties
data class Badge(
    var name: String = "",
    var description: String = "",
    var link: String = ""
)

@IgnoreExtraProperties
data class SocialAccount(
    var name: String = "",
    var icon: String = "",
    var link: String = ""
)
